app/ilmoweb/templates/open_labs.html

```diff
@@ -110,14 +112,29 @@ <h1 class="modal-title fs-5" id="studentsLabel">
              </div>
             </div>
            </td>
+          {% elif lab_group.signed_up_students == lab.max_students %}
```

Comment:
I think this is not enough to make sure the group size is not too big. The method for signing up should check the value from the database since multiple students might have this page open at the same time.